### PR TITLE
Rank Pairing Heap Implementation 

### PR DESCRIPTION
**Pull request by erjiaqing**

### CODE REVIEW

File: rank_pairing/rank_pairing_heap.go [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+	Val                heap.Item
+	Left, Next, Parent *node
+	Rank               int
+}
+
+// RPHeap is an implementation of a Rank Pairing Heap.
+// The zero value for RPHeap Root is an empty Heap.
+type RPHeap struct {
+	head *node
+	size int
```
Comment (theodesp) [inline rank_pairing/rank_pairing_heap.go]: We don't really use size in all the heaps so far

File: rank_pairing/rank_pairing_heap.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
+package rank_paring
+
+import (
+	"fmt"
+
+	heap "github.com/theodesp/go-heaps"
+)
+
+type node struct {
+	Val                heap.Item
```
Comment (theodesp) [inline rank_pairing/rank_pairing_heap.go]: Can you rename it to `item` ?

File: rank_pairing/rank_pairing_heap.go [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+		if ptr != nil {
+			r.insertRoot(ptr)
+		}
+	}
+	return ret
+}
+
+// Clear the whole RankPairingHeap
+func (r *RPHeap) Clear() {
+	r.head = nil
```
Comment (theodesp) [inline rank_pairing/rank_pairing_heap.go]: We want head to contain an empty node

File: rank_pairing/rank_pairing_heap.go (hunk 1 above)
Comment (erjiaqing) [reply rank_pairing/rank_pairing_heap.go]: size is used in DeleteMin()

**Comment by theodesp** [comment]
Ta

**Pull request merged**